stdlib/public/core/UnicodeSPI.swift

```diff
@@ -199,3 +199,32 @@ extension Unicode.Scalar.Properties {
     return result
   }
 }
+
+//===----------------------------------------------------------------------===//
+// String Word Breaking
+//===----------------------------------------------------------------------===//
+
+extension String {
+  @_spi(_Unicode)
+  @available(SwiftStdlib 5.7, *)
+  public func _wordIndex(after i: String.Index) -> String.Index {
+    _precondition(i < endIndex)
+
+    let i = _guts.validateWordIndex(i)
```

Comment:
The preconditon isn't necessary now that we have the explicit index validation call immediately below.

```suggestion
    let i = _guts.validateWordIndex(i)
```